wireapp/core-crypto
# feat: Added support for MLS Group persistence [CL-5]

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Upon restarts, CoreCrypto would "forget" which groups the Client was in and their state, leading to needing to get new Welcome messages for no reason.

### Causes (Optional)

MLS Groups were not being persisted on disk

### Solutions

Using the OpenMLS group persistence API (`MlsGroup::load()` / `MlsGroup::save()`)


### Testing

#### Test Coverage (Optional)


#### How to Test

Create an MlsCentral. Create a conversation.
Drop the MlsCentral. Try to encrypt a message for the above conversation ID.
Before: Error saying the conversation doesn't exist.
Now: Works as expected.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
